packages/mobile/src/localCurrency/saga.ts

```diff
@@ -43,23 +45,31 @@ export async function fetchExchangeRate(symbol: string): Promise<number> {
 }
 
 export function* fetchLocalCurrencyRateSaga() {
-  if (!LOCAL_CURRENCY_SYMBOL) {
-    return
-  }
-
   try {
-    const rate = yield call(fetchExchangeRate, LOCAL_CURRENCY_SYMBOL)
-    yield put(fetchCurrentRateSuccess(rate, Date.now()))
+    const localCurrencyCode: LocalCurrencyCode | null = yield select(getLocalCurrencyCode)
+    if (!localCurrencyCode) {
+      throw new Error("Can't fetch local currency rate without a currency code")
+    }
+    const rate = yield call(fetchExchangeRate, localCurrencyCode)
+    yield put(fetchCurrentRateSuccess(localCurrencyCode, rate, Date.now()))
   } catch (error) {
     Logger.error(`${TAG}@fetchLocalCurrencyRateSaga`, error)
```

Comment:
It would be good to `yield put(showError...` here with a custom message

Reply:
Since this is a background task, I think it would be a surprising experience to have an error banner displayed on unrelated screens.

I'm thinking also specifically about the case where the blockchain api endpoint is down and this error would trigger continuously until succeeds. Pretty annoying for the user.
We would need additional logic to ensure we don't display the error too often.

In conclusion, I'd rather avoid that for now and keep it as is.

Reply:
Fair point :)